In [1]:
from sklearn.ensemble import RandomForestClassifier, RandomForestRegressor
from sklearn.model_selection import train_test_split
from sklearn.metrics import classification_report
from sklearn.pipeline import Pipeline
import pandas as pd
import joblib
from sklearn.preprocessing import StandardScaler

In [2]:
df = pd.read_csv('../data/processed/AUS_weather_features.csv')

In [3]:
df

MinTemp  MaxTemp  Rainfall  WindGustSpeed  WindSpeed9am  WindSpeed3pm  \
0          13.4     22.9       0.6           44.0          20.0          24.0   
1           7.4     25.1       0.0           44.0           4.0          22.0   
2          12.9     25.7       0.0           46.0          19.0          26.0   
3           9.2     28.0       0.0           24.0          11.0           9.0   
4          17.5     32.3       1.0           41.0           7.0          20.0   
...         ...      ...       ...            ...           ...           ...   
143026      2.8     23.4       0.0           31.0          13.0          11.0   
143027      3.6     25.3       0.0           22.0          13.0           9.0   
143028      5.4     26.9       0.0           37.0           9.0           9.0   
143029      7.8     27.0       0.0           28.0          13.0           7.0   
143030     14.9     22.6       0.0           39.0          17.0          17.0   

        Humidity9am  Humidity3pm  Pressure9am  Pressure3pm  ...  windGust_x  \
0              71.0         22.0       1007.7       1007.1  ...   -1.000000   
1              44.0         25.0       1010.6       1007.8  ...   -0.923880   
2              38.0         30.0       1007.6       1008.7  ...   -0.923880   
3              45.0         16.0       1017.6       1012.8  ...    0.707107   
4              82.0         33.0       1010.8       1006.0  ...   -1.000000   
...             ...          ...          ...          ...  ...         ...   
143026         51.0         24.0       1024.6       1020.3  ...    1.000000   
143027         56.0         21.0       1023.5       1019.1  ...   -0.382683   
143028         53.0         24.0       1021.0       1016.8  ...    0.000000   
143029         51.0         24.0       1019.4       1016.5  ...    0.707107   
143030         62.0         36.0       1020.2       1017.9  ...    1.000000   

          windGust_y  wind9am_x     wind9am_y  wind3pm_x     wind3pm_y  Year  \
0      -1.836970e-16  -1.000000 -1.836970e-16  -0.923880  3.826834e-01  2008   
1       3.826834e-01  -0.382683  9.238795e-01  -0.923880 -3.826834e-01  2008   
2      -3.826834e-01  -1.000000 -1.836970e-16  -0.923880 -3.826834e-01  2008   
3       7.071068e-01   0.707107 -7.071068e-01   1.000000  6.123234e-17  2008   
4      -1.836970e-16   0.923880  3.826834e-01  -0.707107  7.071068e-01  2008   
...              ...        ...           ...        ...           ...   ...   
143026  6.123234e-17   0.707107 -7.071068e-01   0.923880  3.826834e-01  2017   
143027  9.238795e-01   0.707107 -7.071068e-01   0.000000  1.000000e+00  2017   
143028  1.000000e+00   0.707107 -7.071068e-01  -0.923880  3.826834e-01  2017   
143029 -7.071068e-01   0.382683 -9.238795e-01   0.000000  1.000000e+00  2017   
143030  6.123234e-17   0.923880 -3.826834e-01   0.923880 -3.826834e-01  2017   

        Month  Day  DayOfYear  
0          12    1        336  
1          12    2        337  
2          12    3        338  
3          12    4        339  
4          12    5        340  
...       ...  ...        ...  
143026      6   21        172  
143027      6   22        173  
143028      6   23        174  
143029      6   24        175  
143030      6   25        176  

[143031 rows x 24 columns]

In [4]:
X = df.drop(columns=['RainTomorrow'])
y_class = df['RainTomorrow']

x_train, x_test, y_class_train, y_class_test = train_test_split(
    X, y_class, test_size= 0.2, random_state=6
)

In [5]:
X.columns

Index(['MinTemp', 'MaxTemp', 'Rainfall', 'WindGustSpeed', 'WindSpeed9am',
       'WindSpeed3pm', 'Humidity9am', 'Humidity3pm', 'Pressure9am',
       'Pressure3pm', 'Temp9am', 'Temp3pm', 'RainToday', 'windGust_x',
       'windGust_y', 'wind9am_x', 'wind9am_y', 'wind3pm_x', 'wind3pm_y',
       'Year', 'Month', 'Day', 'DayOfYear'],
      dtype='object')

import joblib
import pandas as pd

# Load model
model = joblib.load("model.pkl")

# Ambil 1 sampel data hari ini
today_data = pd.DataFrame({
    'MinTemp': [22.4],
    'MaxTemp': [30.5],
    'Rainfall': [0.2],
    'Humidity3pm': [65],
    'Pressure3pm': [1013],
    'Temp3pm': [29.1],
    'RainToday': [0],
    'WindSpeed3pm': [15],
    # dst...
})

# Prediksi biner
prediction = model.predict(today_data)[0]

# Prediksi probabilitas
rain_prob = model.predict_proba(today_data)[0][1]  # kolom ke-1 = probabilitas hujan

print(f"Besok kemungkinan hujan: {rain_prob*100:.2f}%")
